frontend/admin/src/js/components/useLocale.tsx

```diff
@@ -0,0 +1,9 @@
+import { useIntl } from "react-intl";
+import { getLocale } from "@common/helpers/localize";
+
+export default function useLocale() {
```

Comment:
Love this! :)

Reply:
Needing to import and deconstruct `useIntl` just to get the locale and never use `intl` for any other reason is annoying.

Ideally we move the logic in `getLocale` here and phase it out.

```tsx

const useLocale = () => {
    const { intl: { locale } } = useIntl();

    if (locale === "en" || locale === "fr") {
      return locale;
    }

    return "en";
}

Reply:
Yup, it is a very common pattern to do:

```tsx
const useXContext = () => {
  const context = React.useContext(XContext);
  
  return context;
}
```

Reply:
My only suggestion is to move this useLocale hook to /common

Reply:
Done in 8976c920a

Spun out issue for larger effort: https://github.com/GCTC-NTGC/gc-digital-talent/issues/3677